pkgs/tools/inputmethods/libinput-gestures/default.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, fetchFromGitHub, libinput, xdotool, makeWrapper }:
+stdenv.mkDerivation rec {
+  pname = "libinput-gestures";
+  version = "2.32";
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "bulletmark";
+    repo = "libinput-gestures";
+    rev = version;
+    sha256 = "1by6sabx0s8sd9w5675gc26q7yccxnxxsjg4dqlb6nbs0vcg81s7";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase =
+    ''
+      mkdir -p $out
+      $shell libinput-gestures-setup -d "$out" install
+      mv "$out/usr/bin" "$out/bin-unwrapped"
+      mv "$out/usr/share" "$out/share"
+      rmdir "$out/usr"
+      mkdir "$out/bin"
+      substituteInPlace "$out/share/applications/libinput-gestures.desktop" --replace "/usr" "$out"
+      chmod +x "$out/share/applications/libinput-gestures.desktop"
```

Comment:
.desktop files without a shebang shouldn't be executable.

Reply:
I agree, but KDE doesn't seem to. If it's not there then running `kioclient5 exec result/share/applications/libinput-gestures.desktop` gives the following confirmation prompt:

![image](https://user-images.githubusercontent.com/649832/36402597-fd73c378-15df-11e8-8d42-ad8f3fa8b301.png)

Clicking Continue fails with the following message:

![image](https://user-images.githubusercontent.com/649832/36402605-1457c6a2-15e0-11e8-9aab-867b53166675.png)

Since /nix/store is read-only. If `libinput-gestures.desktop` is executable then everything works fine.

Reply:
FWIW, Firefox seems to [`install` its .desktop file](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/browsers/firefox/wrapper.nix#L138), which also causes it to become marked executable.

Reply:
I did not know that. Thanks!

Reply:
Yeah.. I was surprised too when I found out.